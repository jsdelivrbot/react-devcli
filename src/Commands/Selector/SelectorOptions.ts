export const name = 'selector';

export const description = 'Selector description';

export const flags = {};
