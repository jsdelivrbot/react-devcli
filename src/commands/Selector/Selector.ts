import {Arguments} from 'yargs';
import ACommand from '../Command/ACommand';
import {IMakeable} from '../Command/contracts/IMakeable';
import {IRemovable} from '../Command/contracts/IRemovable';
import {ITestable} from '../Command/contracts/ITestable';
import {SelectorsConfig} from 'react-devcli';

class Selector extends ACommand implements IRemovable, ITestable, IMakeable {
    public constructor() {
        const name = 'reducer';
        const description = 'Reducer description';
        const options = {
        };

        super(name, description, options);
    }

    public run(argv: Arguments, config: SelectorsConfig): void {
        console.log(argv);
        console.log(config);
    }

    public help(argv: Arguments): void {
        console.log(argv);
    }

    public remove(argv: Arguments): void {
        console.log(argv);
    }

    public test(argv: Arguments): void {
        console.log(argv);
    }

    public make(argv: Arguments): void {
        console.log(argv);
    }
}

export default new Selector();
