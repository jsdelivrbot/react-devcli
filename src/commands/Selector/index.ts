import Selector from './Selector';

export default new Selector();
