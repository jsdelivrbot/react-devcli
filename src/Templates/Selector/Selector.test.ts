import {js, ts} from './Selector';

describe('Selector template test suite', (): void => {
    test('should match selector template snapshot for JS language', (): void => {
        expect(js()).toMatchSnapshot();
    });

    test('should match selector template snapshot for TS language', (): void => {
        expect(ts()).toMatchSnapshot();
    });
});
