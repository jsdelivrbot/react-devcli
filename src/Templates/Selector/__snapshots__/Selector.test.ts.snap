// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Selector template test suite should match selector template snapshot for JS language 1`] = `""`;

exports[`Selector template test suite should match selector template snapshot for TS language 1`] = `""`;
